lib/cequel/record/data_set_builder.rb

```diff
@@ -122,6 +122,14 @@ def set_paging_state
       def sort_direction
         ascends_by?(order_by_column) ? :asc : :desc
       end
+
+      def method_missing(m, *args, &block)
+        if record_set_delegated_methods.include?(m)
+          record_set.send(m, *args, &block)
+        else
+          super
+        end
+      end
```

Comment:
`#method_missing` is *much* slower than straight delegation. Why change?

Reply:
If the problem was the warning about delegating to a private method we could convert the target method to `protected`. Protected is generally a better choice than private anyway, IMO.

Reply:
For the schema/table.rb note, it now fails on repeated columns. I added a test to test this case as well.

Reply:
Would have been great to get this feedback about 8 months ago. :-) Alright.. i'll rework the entire approach to try to make it work.  Ugh. :-(

Reply:
Sorry i didn't see this until today. I changes i did request didn't get done until today, though.

If we need access to those method from another class then they should be public. Period. Use `def_delegate` to bypass method access control is a terrible idea. Let's just make them public and call it a day.

Reply:
I guess this boils down to a difference in our respective development cultures. 

I've already, based on your last comment, reimplemented them in a way that doesn't call the method_missing at all or use delegation, by exposing them as method directly on the classes.  I had already started this before I saw your comment above (stayed up late last night to implement this to get this finally done). 

If this is acceptable, then OK. If not, let me know.. as I'm not super excited about making another significant change to the code base. 

On your other comments about changes not being done until yesterday, a few thoughts for the future. 

With all the other changes made, and that **I did go through and remove comments**, but evidently missed a few, I wouldn't have expected that to hold up these changes for as long as it has.  I was under the impression that I had removed all the commented out code. If you were waiting on me to remove the 3 or 4 more lines of code, then it would have been great for you to tell me that, and point me to where you found them for me to remove. Or, even better, just remove them yourself. That "lag" of 8 months over 3 or 4 lines of commented out code seems like a waste.

And, on the other issue (add_column erroring versus taking the last one in), I didn't perceive your comment as an absolute request--more of a question about how it was implemented. We're talking about a method that had 3 or 4 lines of code. If there was a needed change, it wouldn't have been hard to fix.. and all of this could have been in production 8+ months ago. 
Anyway, it's been changed to raise an error now, and I added a test for it to ensure that it does. 

Anyway, it's different culture, and all good. 

Let me know about the delegation stuff.